src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileProperties.java

```diff
@@ -2,7 +2,7 @@
 
 import java.io.File;
 
-public class FileDescriptor {
+public class FileProperties {
```

Comment:
FileProperties is misleading because it is a term usually used to describe stuff like : creation data, version, size and so forth.

TBH, FileDescriptor is also a bad, because it is also reserved in some languages.

I would go with SourceFile